http/file_server.ts

```diff
@@ -233,16 +251,20 @@ function fileLenToString(len: number): string {
  * Returns an HTTP Response with the requested file as the body.
  * @param req The server request context used to cleanup the file handle.
  * @param filePath Path of the file to serve.
+ * @param etagAlgorithm The algorithm to use for generating the ETag. Defaults to "fnv1a".
+ * @param fileInfo An optional FileInfo object returned by Deno.stat. It is used
+ * for optimization purposes.
  */
 export async function serveFile(
   req: Request,
   filePath: string,
+  etagAlgorithm: EtagAlgorithm = "fnv1a",
+  fileInfo?: Deno.FileInfo,
 ): Promise<Response> {
-  const [file, fileInfo] = await Promise.all([
-    Deno.open(filePath),
-    Deno.stat(filePath),
-  ]);
-
+  const file = await Deno.open(filePath);
+  if (fileInfo === undefined) {
+    fileInfo = await Deno.stat(filePath);
+  }
```

Comment:
This serializes the calls. You can still have them be parallel while adding this optimization.